src/AppInstallerCommonCore/Runtime.cpp

```diff
@@ -262,135 +201,86 @@ namespace AppInstaller::Runtime
         return LocIndString{ strstr.str() };
     }
 
-    std::filesystem::path GetPathToTemp()
+    std::filesystem::path GetPathTo(PathName path)
     {
         std::filesystem::path result;
 
         if (IsRunningInPackagedContext())
         {
-            result.assign(winrt::Windows::Storage::ApplicationData::Current().TemporaryFolder().Path().c_str());
-        }
-        else
-        {
-            wchar_t tempPath[MAX_PATH + 1];
-            DWORD tempChars = GetTempPathW(ARRAYSIZE(tempPath), tempPath);
-            result.assign(std::wstring_view{ tempPath, static_cast<size_t>(tempChars) });
-        }
+            auto appStorage = winrt::Windows::Storage::ApplicationData::Current();
 
-        result /= AICLI_DEFAULT_TEMP_DIRECTORY;
-
-        std::filesystem::create_directories(result);
-
-        return result;
-    }
-
-    std::filesystem::path GetPathToLocalState()
-    {
-        if (IsRunningInPackagedContext())
-        {
-            std::filesystem::path result = winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path().c_str();
+            switch (path)
+            {
+            case PathName::Temp:
+                result.assign(appStorage.TemporaryFolder().Path().c_str());
+                result /= AICLI_DEFAULT_TEMP_DIRECTORY;
+                break;
+            case PathName::LocalState:
+            case PathName::UserFileSettings:
+                result.assign(appStorage.LocalFolder().Path().c_str());
 
 #ifndef AICLI_DISABLE_TEST_HOOKS
-            if (!s_Settings_TestHook_ForcedContainerPrepend.empty())
-            {
-                result /= s_Settings_TestHook_ForcedContainerPrepend;
-            }
+                if (!s_Settings_TestHook_ForcedContainerPrepend.empty())
+                {
+                    result /= s_Settings_TestHook_ForcedContainerPrepend;
+                }
 #endif
-
-            return result;
-        }
-        else
-        {
-            return GetPathToAppDataDir(s_AppDataDir_State);
-        }
-    }
-
-    std::filesystem::path GetPathToDefaultLogLocation()
-    {
-        if (IsRunningInPackagedContext())
-        {
-            // To enable UIF collection through Feedback hub, we must put our logs here.
-            auto result = GetPathToLocalState() / WINGET_DEFAULT_LOG_DIRECTORY;
-
-            std::filesystem::create_directories(result);
-
-            return result;
+                break;
+            case PathName::DefaultLogLocation:
+                // To enable UIF collection through Feedback hub, we must put our logs here.
+                result.assign(appStorage.LocalFolder().Path().c_str());
+                result /= WINGET_DEFAULT_LOG_DIRECTORY;
+                break;
+            case PathName::StandardSettings:
+#ifndef AICLI_DISABLE_TEST_HOOKS
+                result = s_Settings_TestHook_ForcedContainerPrepend;

```

Comment:
>result = s_Settings_TestHook_ForcedContainerPrepend; [](start = 16, length = 52)

Do we need the empty check if (!s_Settings_TestHook_ForcedContainerPrepend.empty()) here as above?

Reply:
No.  We probably don't need the other one, but I feel safer assigning an empty one than appending.